source/MediaPlayerManager.cpp

```diff
@@ -104,9 +104,15 @@ void MediaPlayerManager::releaseMediaPlayerClient()
             it->second.refCount--;
             if (it->second.refCount == 0)
             {
-                it->second.client->stop();
-                it->second.client->stopStreaming();
-                it->second.client->destroyClientBackend();
+                auto client = it->second.client;
+                client->stop();
+
+                // destroyClientBackend() unsubscribes from messages. Call this before
+                // calling stopStreaming() otherwise received messages can't be sent
+                // and this results in an ERROR message in the log
+                client->destroyClientBackend();
```

Comment:
I dont like the idea of resetting clientBackend before stopping the backendQueue, is it now possible that we could receive/handle another message before backendqueue is stopped? In that case we would get segmentation fault.